### DISCUSSION TITLE
如何实现Pot播放器的画面重置/恢复功能？

### DISCUSSION THREAD

**Discussion by Trudeau566** [hooke007/mpv_PlayKit#524] [mpv 通用]
比如我放大了视频画面，然后又重置为零，怎么才能实现恢复到我之前放大的尺寸？
Pot播放器有这样的功能，mpv貌似没有？
![PixPin_2024-11-30_12-05-57](https://github.com/user-attachments/assets/4158b9ed-2305-4019-945e-be2bb3a47e3e)

**Comment by hooke007** [comment]
这东西写起来也不难。
参考
https://github.com/hooke007/MPV_lazy/blob/445d07cc5a933d2fee4bb20d54900b253da747c5/portable_config/scripts/input_plus.lua#L60

**Reply by Trudeau566** [reply]
> 这东西写起来也不难。 参考
> 
> https://github.com/hooke007/MPV_lazy/blob/445d07cc5a933d2fee4bb20d54900b253da747c5/portable_config/scripts/input_plus.lua#L60

大佬我研究了一下午，终于实现了video-zoom的重置与恢复，但是我想用一个快捷键实现video-zoom加video-pan-y加video-pan-x的重置与恢复，不知道怎么搞
这是我自己改的您看下该怎么添加？
function speed_recover()
	if mp.get_property_number("video-zoom") ~= 0 then
		bak_speed = mp.get_property_number("video-zoom")
		mp.command("set video-zoom 0")
	else
		if bak_speed == nil then bak_speed = 1 end
		mp.command("set video-zoom " .. bak_speed)
	end
end

**Reply by hooke007** [reply]
不感兴趣，可以让其它感兴趣的热心人解答

**Comment by Trudeau566** [comment]
经过几天的研究终于搞成了，完全复原了pot的画面重置/恢复的效果

https://github.com/user-attachments/assets/3a9aa1b1-9608-4079-a4d3-f1cae60e23eb
